### PR TITLE
fix(core): prevent endless reload on cache requests

### PR DESCRIPTION
**Pull request by snnd**
Excuse me in advance for writing a bit of a long story. The source of the error isn't quite obvious so I wanted to explain the szenario in detail. There is a TLDR at the end. of this PR.

So, this error popped up after I extracted the "needed files" modal dialog to a separate component. At first I was wondering why the bug did not pop up earlier. This is most likely due to a "race condition alike" situation. It is caused by the vendor/angular/angular-dirPagination.js component. Usually the directive gets rendered/executed, when window.guiVersion has been initialized by one of the previous requests, whose respose carries 'x-syncthing-version'. angular-dirPagination itself makes use of HTML templates, which are embedded as a JS string and stored into angulars $templateCache. This is a very common strategy for angular directives, since one can avoid many problems while handling external views. Syncthing's UI components don't need this technique since we don't concat/minimize the webgui sources yet and actually since some months ago, had all the HTML in just a single file. 

As I mentioned in my modal refactoring PR, here is the proposal for a particular bugfix. It never the less actually feels a bit more like symptom fixing. I think the idea, to continuesly track the GUI version number and if necessary force reload the GUI, might be the conceptual error here (race conditions, empty headers, etc). Also this $templateCache technique can come up in future development, when we integrate more third party libraries or decide to concat and minimize our webgui sources.

TLDR: Anyhow after the refactoring, the pagination got rendered earlier, makes a kind of fake HTTP request to the templateCache and our custom HTTP Interceptor finds no header in the reponse. So it runs into the errornous case and hangs up into a reload loop. So I symptom fixed this by skipping the guiVersion validation when there are no headers in the response, which actually only should happen in case of a manually stored $templateCache item.

If anyone can offer a better solution here, I would be pleased to know about your idea. :)

### CODE REVIEW

**Comment by AudriusButkevicius** [comment]
So I am super surprised that some template lookup mechanism uses http interceptors, that seems like a flaw in angular, though the fix is reasonable. I guess we could check for `undefined` and bomb out early. 

We could also invert the if condition to reduce the indentation

```
if (Object.keys(headers).length == 0) 
   return response;

break_stuff()
```

**Comment by snnd** [comment]
I did some research and yes, actually it seems to be an architectural dilemma at angularjs, as I interpreted the discussion at https://github.com/angular/angular.js/issues/10257

I am in for bombing our early and strongly disencourage using non strict comparision/evaluation since to JavaScripts type coercion.

I also improved the code here a bit. For instnace I eleminated the global state of window.guiVersion and window.deviceId since, after searching globally for " NAME" app.js is the only occourence. Since the httpInterceptor is a singleton, the variable will be held within the interceptor's scope.

**Comment by snnd** [comment]
I would also be in for extracting the interceptor to a seperate configuration file.
